### PR TITLE
W-12161504: Fix Cookie replacement during redirect mechanism

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java [hunk 1]
```diff
@@ -172,6 +185,96 @@ public void caseSensitivity() {
...
+  @Issue("W-12161504")
+  public void setCookieWithDateInThePastRemovesIt() {
+    RedirectUtils redirectUtils = new RedirectUtils(false, false);
+    addSetCookieHeader("TheCookieName=TheCookieValue");
+    addSetCookieHeader("TheCookieName=NewCookieValue; Expires=Sat, 02 Oct 1993 14:20:00 GMT");
+
+    RequestBuilder requestBuilder = new RequestBuilder();
+    redirectUtils.handleResponseCookies(requestBuilder, response);
+    Collection<String> cookiesInRequestAsString =
+        requestBuilder.build().getCookies().stream().map(Cookie::toString).collect(Collectors.toList());
```
Comment (jcgrethe) [inline src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java:270]: static import

File: src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java [hunk 2]
```diff
@@ -172,6 +185,96 @@ public void caseSensitivity() {
...
+  @Issue("W-12161504")
+  public void setTwoCookiesWithSameName() {
+    RedirectUtils redirectUtils = new RedirectUtils(false, false);
+    addSetCookieHeader("TheCookieName=OldCookieValue");
+    addSetCookieHeader("TheCookieName=NewCookieValue");
+
+    RequestBuilder requestBuilder = new RequestBuilder();
+    redirectUtils.handleResponseCookies(requestBuilder, response);
+    Collection<String> cookiesInRequestAsString =
+        requestBuilder.build().getCookies().stream().map(Cookie::toString).collect(Collectors.toList());
```
Comment (jcgrethe) [inline src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java:256]: static import

File: src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java [hunk 3]
```diff
@@ -172,6 +185,96 @@ public void caseSensitivity() {
...
+  @Test
+  @Issue("W-12161504")
+  public void setCookieHeaderInResponseIsAddedAsCookieInNextRequest() {
+    RedirectUtils redirectUtils = new RedirectUtils(false, false);
+    addSetCookieHeader("TheCookieName=TheCookieValue");
+
+    RequestBuilder requestBuilder = new RequestBuilder();
+    redirectUtils.handleResponseCookies(requestBuilder, response);
+    Collection<String> cookiesInRequestAsString =
+        requestBuilder.build().getCookies().stream().map(Cookie::toString).collect(Collectors.toList());
```
Comment (jcgrethe) [inline src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java:242]: static import

**Review by jcgrethe — CHANGES_REQUESTED** [review]

**Review by jcgrethe — APPROVED** [review]

**Comment by eze210** [comment]
--validate

**Pull request merged**